src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1233,7 +1238,10 @@ else if (count <= fanout * controller.getFanout(index + 1) || maxSSTablesToCompa
                                                              arena,
                                                              level);
 
-                CompactionPick pick = CompactionPick.create(index, pullOldestSSTables(maxSSTablesToCompact));
+                Collection<CompactionSSTable> pickedSSTables = pullOldestSSTables(maxSSTablesToCompact);
+                CompactionPick pick = CompactionPick.create(index, pickedSSTables);
+                int updatedMaxOverlap = pickedSSTables.size();
```

Comment:
`maxSSTablesToCompact` here refers to maximum number of _overlapping_ sstables, while this returns the total number of sstables picked.

Reply:
reverted